flocker/common/thread.py

```diff
@@ -0,0 +1,67 @@
+# Copyright ClusterHQ Inc.  See LICENSE file for details.
+
+"""
+Some thread-related tools.
+"""
+
+from twisted.internet.threads import deferToThreadPool
+
+from interface import interface_decorator
```

Comment:
This should be an "explicit relative import" instead of an "implicit relative import" (or, if Flocker didn't use explicit relative imports all over the place, it should have been an absolute import instead of an implicit relative import - but we do so ...).

Relative imports are ridiculously error prone.  They're a horrible misfeature of the language.  Just one of the terrible side-effects of doing this is that afterwards there are actually two copies of the interface module loaded.  One of them is named `flocker.common.interface` and another is called `interface`.  And they each define objects that look the same but they're actually distinct objects in distinct modules and this can lead to all kinds of nastiness...

So `from .interface import interface_decorator`.  And then, ideally, after `interface` becomes `_interface` and its function gets re-exposed in the `flocker.common` module, `from . import interface_decorator` - but you won't actually be able to do this since it will create a circular dependency between `flocker/common/__init__.py` and `flocker/common/thread.py`.  Very sad.  This is the main reason I'm not sure I like the idiom we use of having a bunch of private modules and then a `__init__.py` that imports a bunch of stuff from them.